editor/src/components/inspector/flex-section.tsx

```diff
@@ -106,10 +122,80 @@ export const FlexSection = React.memo(() => {
     [addFlexLayoutSystem, addGridLayoutSystem],
   )
 
+  const grid = useEditorState(
+    Substores.metadata,
+    (store) =>
+      layoutSystem === 'grid' && store.editor.selectedViews.length === 1
+        ? MetadataUtils.findElementByElementPath(
+            store.editor.jsxMetadata,
+            store.editor.selectedViews[0],
+          )
+        : null,
+    'FlexSection grid',
+  )
+
+  const columns: GridCSSNumber[] = React.useMemo(() => {
+    if (grid == null) {
+      return []
+    }
+
+    const fromProps =
+      grid.specialSizeMeasurements.containerGridPropertiesFromProps.gridTemplateColumns
+    const calculated = grid.specialSizeMeasurements.containerGridProperties.gridTemplateColumns
+
+    if (fromProps?.type !== 'DIMENSIONS' && calculated?.type !== 'DIMENSIONS') {
+      return []
+    }
+
+    let dimensions: GridCSSNumber[] = []
+
+    if (calculated?.type === 'DIMENSIONS') {
+      dimensions.push(...calculated.dimensions)
+    }
+    if (fromProps?.type === 'DIMENSIONS') {
+      if (dimensions.length === 0) {
+        return fromProps.dimensions
+      } else if (dimensions.length === fromProps.dimensions.length) {
+        return fromProps.dimensions
+      }
+    }
+
+    return dimensions
+  }, [grid])
+
+  const rows: GridCSSNumber[] = React.useMemo(() => {
+    if (grid == null) {
+      return []
+    }
+
+    const fromProps = grid.specialSizeMeasurements.containerGridPropertiesFromProps.gridTemplateRows
+    const calculated = grid.specialSizeMeasurements.containerGridProperties.gridTemplateRows
+
+    if (fromProps?.type !== 'DIMENSIONS' && calculated?.type === 'DIMENSIONS') {
+      return []
+    }
+
+    let merged: GridCSSNumber[] = []
+
+    if (calculated?.type === 'DIMENSIONS') {
+      merged.push(...calculated.dimensions)
+    }
+
+    if (fromProps?.type === 'DIMENSIONS') {
+      if (merged.length === 0) {
+        return fromProps.dimensions
+      } else if (merged.length === fromProps.dimensions.length) {
+        return fromProps.dimensions
+      }
+    }
+
+    return merged
```

Comment:
looks like a copy paste of the above useMemo, it could be extracted into a function where fromProps and calculated are the parameters (or a parameter whether it is row or column)

Reply:
good point! https://github.com/concrete-utopia/utopia/pull/6118/commits/b7cfd566e9c4781df314c0c90f8f92e70c8b2e67